x-pack/plugins/lens/server/usage/collectors.ts

```diff
@@ -69,24 +67,10 @@ function addEvents(prevEvents: Record<string, number>, newEvents: Record<string,
 }
 
 async function getLatestTaskState(taskManager: TaskManagerStartContract) {
-  try {
-    const result = await taskManager.fetch({
-      query: { bool: { filter: { term: { _id: `task:Lens-lens_telemetry` } } } },
-    });
-    return result.docs;
-  } catch (err) {
-    const errMessage = err && err.message ? err.message : err.toString();
-    /*
-      The usage service WILL to try to fetch from this collector before the task manager has been initialized, because the
-      task manager has to wait for all plugins to initialize first. It's fine to ignore it as next time around it will be
-      initialized (or it will throw a different type of error)
-    */
-    if (!errMessage.includes('NotInitialized')) {
-      throw err;
-    }
-  }
-
-  return null;
+  const result = await taskManager.fetch({
+    query: { bool: { filter: { term: { _id: `task:Lens-lens_telemetry` } } } },
+  });
+  return result.docs;
```

Comment:
Why removing the `try/catch` here?
It seems unrelated to the `isReady` change

Reply:
The try catch was only necessary because early task manager didn't have an `isReady` check. As that's in place now, we don't need to fake everything's working fine (which is what the catch was doing here).

Reply:
I see what you mean, but when calling this function there was already an `await taskManager` here: https://github.com/elastic/kibana/pull/89109/files#diff-baaab05f5a10da483776bde09fa1a8aefa3f83ba17db0b437a630bf1fc70330fR23

So probably the `try/catch` had no sense before as well : `await taskManager` is the whole `isReady` logic.

Reply:
Looking into the commit where this got introduced I think you are right, I remembered it the wrong way. Reintroduced the catch.